lib/net.js

```diff
@@ -825,7 +825,17 @@ protoGetter('bytesWritten', function bytesWritten() {
       bytes += Buffer.byteLength(el.chunk, el.encoding);
   });
 
-  if (data) {
+  if (Array.isArray(data)) {
+    // was a writev, iterate over chunks to get total length
+    for (var i = 0; i < data.length; i++) {
+      const chunk = data[i];
+
+      if (chunk instanceof Buffer)
```

Comment:
I've added the `data.allBuffers` check, thanks.

@addaleax I'm not so sure that's faster — [this benchmark](https://gist.github.com/brendanashworth/3caa0d50b14819863e534ea9bfdbbe72) says its a bit slower, probably because of going past the C++ boundary.